dockerfile/parser/parser.go

```diff
@@ -312,6 +316,11 @@ func Parse(rwc io.Reader) (*Result, error) {
 		root.AddChild(child, startLine, currentLine)
 	}
 
+	scannerErr := scanner.Err()
+	if scannerErr != nil {
```

Comment:
Go ahead and use the `if scannerErr := scanner.Err(); scannerErr != nil {` form here, since `scannerErr` isn't going to be used anywhere else.